cpp/src/merge/merge.cu

```diff
@@ -0,0 +1,458 @@
+    const cudf::size_type total_size = left_size + right_size;
+
+    thrust::constant_iterator<side> left_side(side::LEFT);
+    thrust::constant_iterator<side> right_side(side::RIGHT);
+
+    auto left_indices = thrust::make_counting_iterator(static_cast<cudf::size_type>(0));
+    auto right_indices = thrust::make_counting_iterator(static_cast<cudf::size_type>(0));
+
+    auto left_begin_zip_iterator = thrust::make_zip_iterator(thrust::make_tuple(left_side, left_indices));
+    auto right_begin_zip_iterator = thrust::make_zip_iterator(thrust::make_tuple(right_side, right_indices));
+
+    auto left_end_zip_iterator = thrust::make_zip_iterator(thrust::make_tuple(left_side + left_size, left_indices + left_size));
+    auto right_end_zip_iterator = thrust::make_zip_iterator(thrust::make_tuple(right_side + right_size, right_indices + right_size));
+
+    rmm::device_vector<index_type> merged_indices(total_size);
+    
+    auto lhs_device_view = cudf::table_device_view::create(left_table, stream);
+    auto rhs_device_view = cudf::table_device_view::create(right_table, stream);
+
+    rmm::device_vector<cudf::order> d_column_order(column_order); 
+    
+    auto exec_pol = rmm::exec_policy(stream);
+    if (nullable){
+      rmm::device_vector<cudf::null_order> d_null_precedence(null_precedence);
+      
+      auto ineq_op =
+        cudf::experimental::detail::row_lexicographic_tagged_comparator<true>(*lhs_device_view,
+                                                                              *rhs_device_view,
+                                                                              d_column_order.data().get(),
+                                                                              d_null_precedence.data().get());
+      
+        thrust::merge(exec_pol->on(stream),
+                    left_begin_zip_iterator,
+                    left_end_zip_iterator,
+                    right_begin_zip_iterator,
+                    right_end_zip_iterator,
+                    merged_indices.begin(),
+                      [ineq_op] __device__ (index_type const & left_tuple,
+                                            index_type const & right_tuple) {
+                        return ineq_op(left_tuple, right_tuple);
+                    });			        
+    } else {
+      auto ineq_op =
+        cudf::experimental::detail::row_lexicographic_tagged_comparator<false>(*lhs_device_view,
+                                                                               *rhs_device_view,
+                                                                               d_column_order.data().get()); 
+        thrust::merge(exec_pol->on(stream),
+                    left_begin_zip_iterator,
+                    left_end_zip_iterator,
+                    right_begin_zip_iterator,
+                    right_end_zip_iterator,
+                    merged_indices.begin(),
+                      [ineq_op] __device__ (index_type const & left_tuple,
```

Comment:
Why does there need to be a lambda here? It looks like you could just use `ineq_op` directly.

Reply:
Fixed.